Toolbar Icons not showing after update (1.6.2 -> 1.8.3) when exposing in a subdir via a reverse proxy and minify = true

![image](https://user-images.githubusercontent.com/16946810/80459501-be29c500-8932-11ea-92eb-8788ceb3c96c.png)

The toolbar Icons are not showing and I get missing font family errors, but they're all in their folders, nothing was changed. Can you help me fix this issue? 
 
GET <server>/static/font/Roboto-Regular.ttf net::ERR_ABORTED 404 (Not Found)

I get this for all font families.

As a reminder :
 
> On the other hand, interesting thing, when I access the service without using the proxy and without https, no problem, the icons are displayed correctly !!

In my case, it's obvious it's proxy related.

In addition, I am in a migration situation of my Etherpad instance and I migrated my database and after that I updated Etherpad. I have this coming out in the logs now:

`[2020-04-29 10:43:29.440] [ERROR] console - table is not configured with charset utf8mb4 -- This may lead to crashes when certain characters are pasted in pads
`
Don't know if it can cause errors…

> As a reminder :
>  
> > On the other hand, interesting thing, when I access the service without using the proxy and without https, no problem, the icons are displayed correctly !!
> 
> In my case, it's obvious it's proxy related.
> 
> In addition, I am in a migration situation of my Etherpad instance and I migrated my database and after that I updated Etherpad. I have this coming out in the logs now:
> 
> `[2020-04-29 10:43:29.440] [ERROR] console - table is not configured with charset utf8mb4 -- This may lead to crashes when certain characters are pasted in pads
> `
> Don't know if it can cause errors…
> 
> 
> 
> 

Create a separate issue for this please.  It needs addressing.

--- a/src/node/handler/PadMessageHandler.js
+++ b/src/node/handler/PadMessageHandler.js
@@ -1076,6 +1076,7 @@ async function handleClientReady(client, message)
     var clientVars = {
       "skinName": settings.skinName,
       "skinVariants": settings.skinVariants,
+      "randomVersionString": settings.randomVersionString,
       "accountPrivs": {
           "maxRevisions": 100
       },

--- a/src/node/utils/Minify.js
+++ b/src/node/utils/Minify.js
@@ -426,17 +426,17 @@ function compressCSS(filename, content, callback)
     /*
      * Changes done to migrate CleanCSS 3.x -> 4.x:
      *
-     * 1. Disabling rebase is necessary because otherwise the URLs for the web
-     *    fonts become wrong.
+     * 1. Rework the rebase logic, because the API was simplified (but we have
+     *    less control now). See:
+     *    https://github.com/jakubpawlowicz/clean-css/blob/08f3a74925524d30bbe7ac450979de0a8a9e54b2/README.md#important-40-breaking-changes
      *
-     *    EXAMPLE 1:
-     *        /static/css/src/static/font/fontawesome-etherpad.woff
-     *      instead of
-     *        /static/font/fontawesome-etherpad.woff
-     *    EXAMPLE 2 (this is more surprising):
-     *        /p/src/static/font/opendyslexic.otf
-     *      instead of
-     *        /static/font/opendyslexic.otf
+     *    EXAMPLE:
+     *        The URLs contained in a CSS file (including all the stylesheets
+     *        imported by it) residing on disk at:
+     *            /home/muxator/etherpad/src/static/css/pad.css
+     *
+     *        Will be rewritten rebasing them to:
+     *            /home/muxator/etherpad/src/static/css
      *
      * 2. CleanCSS.minify() can either receive a string containing the CSS, or
      *    an array of strings. In that case each array element is interpreted as
@@ -447,7 +447,13 @@ function compressCSS(filename, content, callback)
      *    "content" argument, but we have to wrap the absolute path to the CSS
      *    in an array and ask the library to read it by itself.
      */
-    new CleanCSS({rebase: false}).minify([absPath], function (errors, minified) {
+
+    const basePath = path.dirname(absPath);
+
+    new CleanCSS({
+      rebase: true,
+      rebaseTo: basePath,
+    }).minify([absPath], function (errors, minified) {
       if (errors) {
         // on error, just yield the un-minified original, but write a log message
         console.error(`CleanCSS.minify() returned an error on ${filename} (${absPath}): ${errors}`);

--- a/src/static/js/ace.js
+++ b/src/static/js/ace.js
@@ -237,7 +237,7 @@ function Ace2Editor()
 
       // disableCustomScriptsAndStyles can be used to disable loading of custom scripts
       if(!clientVars.disableCustomScriptsAndStyles){
-        $$INCLUDE_CSS("../static/css/pad.css");
+        $$INCLUDE_CSS("../static/css/pad.css?v=" + clientVars.randomVersionString);
       }
 
       var additionalCSS = _(hooks.callAll("aceEditorCSS")).map(function(path){
@@ -247,7 +247,7 @@ function Ace2Editor()
         return '../static/plugins/' + path;
       });
       includedCSS = includedCSS.concat(additionalCSS);
-      $$INCLUDE_CSS("../static/skins/" + clientVars.skinName + "/pad.css");
+      $$INCLUDE_CSS("../static/skins/" + clientVars.skinName + "/pad.css?v=" + clientVars.randomVersionString);
 
       pushStyleTagsFor(iframeHTML, includedCSS);
 
@@ -321,7 +321,7 @@ window.onload = function () {\n\
       var includedCSS = [];
       var $$INCLUDE_CSS = function(filename) {includedCSS.push(filename)};
       $$INCLUDE_CSS("../static/css/iframe_editor.css");
-      $$INCLUDE_CSS("../static/css/pad.css");
+      $$INCLUDE_CSS("../static/css/pad.css?v=" + clientVars.randomVersionString);
 
 
       var additionalCSS = _(hooks.callAll("aceEditorCSS")).map(function(path){
@@ -331,7 +331,7 @@ window.onload = function () {\n\
         return '../static/plugins/' + path }
       );
       includedCSS = includedCSS.concat(additionalCSS);
-      $$INCLUDE_CSS("../static/skins/" + clientVars.skinName + "/pad.css");
+      $$INCLUDE_CSS("../static/skins/" + clientVars.skinName + "/pad.css?v=" + clientVars.randomVersionString);
 
       pushStyleTagsFor(outerHTML, includedCSS);
 

